torch/_dynamo/variables/higher_order_ops.py

```diff
@@ -1127,6 +1127,16 @@ def create_unbacked_sym_node_var(tx) -> SymNodeVariable:
                 unimplemented(
                     f"Expected cond_fn to return a scalar tensor or a bool but got {cond_r_meta.shape}"
                 )
+        elif isinstance(cond_r, ConstantVariable):
+            # short-circuiting while_loop when cond_fn returns a constant such as 0, 1 True or False
```

Comment:
should `while 0` or `while False` still be allowed? (and just do no-op)

Reply:
Yeah, this PR makes it a no-op. The False or 0 could generated by some complicated shape expressions, where corresponding dimension might not be marked as dynamic.